x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java

```diff
@@ -223,34 +229,56 @@ public final class IndexPrivilege extends Privilege {
      * If you are adding a new named index privilege, also add it to the
      * <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/security-privileges.html#privileges-list-indices">docs</a>.
      */
-    private static final Map<String, IndexPrivilege> VALUES = sortByAccessLevel(
-        Stream.of(
-            entry("none", NONE),
-            entry("all", ALL),
-            entry("manage", MANAGE),
-            entry("create_index", CREATE_INDEX),
-            entry("monitor", MONITOR),
-            entry("read", READ),
-            entry("index", INDEX),
-            entry("delete", DELETE),
-            entry("write", WRITE),
-            entry("create", CREATE),
-            entry("create_doc", CREATE_DOC),
-            entry("delete_index", DELETE_INDEX),
-            entry("view_index_metadata", VIEW_METADATA),
-            entry("read_cross_cluster", READ_CROSS_CLUSTER),
-            entry("manage_follow_index", MANAGE_FOLLOW_INDEX),
-            entry("manage_leader_index", MANAGE_LEADER_INDEX),
-            entry("manage_ilm", MANAGE_ILM),
-            entry("manage_data_stream_lifecycle", MANAGE_DATA_STREAM_LIFECYCLE),
-            entry("maintenance", MAINTENANCE),
-            entry("auto_configure", AUTO_CONFIGURE),
-            entry("cross_cluster_replication", CROSS_CLUSTER_REPLICATION),
-            entry("cross_cluster_replication_internal", CROSS_CLUSTER_REPLICATION_INTERNAL),
-            DataStream.isFailureStoreFeatureFlagEnabled() ? entry("read_failure_store", READ_FAILURE_STORE) : null
-        ).filter(Objects::nonNull).collect(Collectors.toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue))
+    private static final Map<String, IndexPrivilege> VALUES = combineSortedInOrder(
```

Comment:
We could for sure clean this up but the all this brouhaha is to account for how `sortByAccessLevel` works:

It counts how many other privileges are a subset of a given privilege and uses that as the "rank" of the privilege.

Given that we now have failure-store specific privileges whose automata are the same as their data-equivalents, this messes up the ordering of data privileges (e.g., read is lower on the scale now).

That's counter-intuitive. What we want instead is to rank failure and data privileges separately. That's what the change below does: rank the separately, then put them back into a sorted map (such that data privileges always come first, then failure privileges).